gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/SpecCatalog.java

```diff
@@ -51,6 +51,8 @@
    * */
```

Comment:
prophetic comment, if there ever was one...

now that you've reworked the metrics gauge, the only remaining use seems to be in managing listeners.  AFAICT (in my budding understanding) that appears reasonable, but might be worth you confirming.

(presuming this optimization bears the copious fruit we hope for) shall we return to update this comment, to scare off casually-considered use?  although perhaps a slight abusage of `@Deprecated` we could shout that caution as a build-time warning.  what's your take?